netenv/addresses.go

```diff
@@ -44,3 +45,48 @@ func GetAssignedGlobalAddresses() (ipv4 []net.IP, ipv6 []net.IP, err error) {
 	}
 	return
 }
+
+var (
+	myIP4s    []net.IP
+	myIP6s    []net.IP
+	myIPsLock sync.Mutex
+)
+
+// IsMyIP returns whether the given IP is currently configured on the local host.
+func IsMyIP(ip net.IP) (yes bool, err error) {
+	if netutils.IPIsLocalhost(ip) {
+		return true, nil
+	}
+
+	if ip4 := ip.To4(); ip4 != nil {
+		return checkMyIP(ip4, func() []net.IP { return myIP4s })
+	}
+	return checkMyIP(ip, func() []net.IP { return myIP6s })
+}
+
+func checkMyIP(ip net.IP, myIPs func() []net.IP) (yes bool, err error) {
+	myIPsLock.Lock()
+	defer myIPsLock.Unlock()
+
+	// check
+	for _, myIP := range myIPs() {
+		if ip.Equal(myIP) {
+			return true, nil
+		}
+	}
+
+	// refresh IPs
+	myIP4s, myIP6s, err = GetAssignedAddresses()
+	if err != nil {
+		return false, err
+	}
+
+	// check again
+	for _, myIP := range myIPs() {
+		if ip.Equal(myIP) {
+			return true, nil
+		}
+	}
```

Comment:
Is that refresh logic really necessary? `checkMyIP` gets a func parameter (`myIPs`) to know against which set of IPs to check but assumes that it's fine to update `myIP4s` and `myIP6s`. What if the function returns something completely different? IMHO that behavior is not clear / mixed a bit. I'd rather drop the refresh part from that function (and move it to the caller if that's really a thing) or replace the `myIPs` parameter with a boolean.
AFAIK there will never be lots of IP addresses so we could actually just use one single slice that has both IPv4 and IPv6 addresses (`ip.Equal()` takes care of that anyway).